soc/arm/nxp_imx/rt6xx/power.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <power/power.h>
+#include "fsl_power.h"
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Invoke Low Power/System Off specific Tasks */
+void pm_power_state_set(struct pm_state_info info)
+{
+	/* FIXME: When this function is entered the Kernel has disabled
+	 * interrupts using BASEPRI register. This is incorrect as it prevents
+	 * waking up from any interrupt which priority is not 0. Work around the
+	 * issue and disable interrupts using PRIMASK register as recommended
+	 * by ARM.
+	 */
+
+	/* Set PRIMASK */
+	__disable_irq();
+	/* Set BASEPRI to 0 */
+	irq_unlock(0);
+
+	switch (info.state) {
+	case PM_STATE_RUNTIME_IDLE:
+		POWER_EnterSleep();
+		break;
```

Comment:
Are you sure that `POWER_EnterSleep` is correctly mapped to `RUNTIME_IDLE` ? This is a question I haven't see this target documentation. Just looking this API name it seems it matches better with `SUSPEND_IDLE`

Reply:
Below is the definition of Sleep mode for this platform:
```
Sleep mode:
Sleep mode saves a significant amount of power by stopping CM33 CPU execution without affecting peripherals or requiring significant wake-up time. The sleep mode affects the relevant CPU only. The clock to the core is shut off. Peripherals and memories are active and operational.
```